packages/scaffold-html/src/modal/w3m-modal/index.ts

```diff
@@ -51,14 +57,64 @@ export class W3mModal extends LitElement {
   }
 
   private async onClose() {
+    this.onScrollUnlock()
     await animate(this, { opacity: [1, 0] }, { duration: 0.2 }).finished
     SnackController.hide()
     this.open = false
+    this.onRemoveKeyboardListener()
   }
 
-  private onOpen() {
+  private async onOpen() {
+    this.onScrollLock()
     this.open = true
-    animate(this, { opacity: [0, 1] }, { duration: 0.2 })
+    await animate(this, { opacity: [0, 1] }, { duration: 0.2 }).finished
+    this.onAddKeyboardListener()
+  }
+
+  private onScrollLock() {
+    const styleTag = document.createElement('style')
+    styleTag.dataset.w3m = SCROLL_LOCK
+    styleTag.textContent = `
+      html, body {
+        touch-action: none;
+        overflow: hidden;
+        overscroll-behavior: contain;
+        scrollbar-gutter: stable;
+      }
+    `
+    document.head.appendChild(styleTag)
+  }
+
+  private onScrollUnlock() {
```

Comment:
I think you will go back to the top of the page if you scroll > then open modal > then close again

Could we set the body `scrollTop` when closing the modal? 

`document.body.style.top =  ${window.scrollY}px;`

Reply:
No need, this works as expected (same implementation as v2) you stay at the same position